### PR TITLE
advanced_login

### PR DESCRIPTION
**Pull request by weilingooi**
advanced login

### CODE REVIEW

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -110,3 +110,18 @@ input.btn.btn-primary {
...
+  span {
+    font-weight: normal;
+    margin-left: 20px;
+  }
+}
+
+#session_remember_me {
+  margin-left: 0;
+  width: auto;
+}
```
Comment (manhnt-0960) [inline app/assets/stylesheets/custom.scss]: add newline at end of file

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -4,15 +4,36 @@ def log_in user
   end
   
   def current_user
-    @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
```
Comment (manhnt-0960) [inline app/helpers/sessions_helper.rb]:
```suggestion
      @current_user ||= User.find_by id: session[:user_id]
```

File: app/models/user.rb [hunk 3]
```diff
@@ -10,13 +11,32 @@ class User < ApplicationRecord
...
+  end
+
+  def authenticated? remember_token
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
+  end
+end
```
Comment (manhnt-0960) [inline app/models/user.rb]: at newline at end of file

**Comment by weilingooi** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**